src/components/edit-file.js

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { Button, Icon } from 'semantic-ui-react'
+
+import EditModal from './edit-modal'
+import { editFile } from '../actions/fileActions'
+import { getFolder } from '../actions/folderActions'
+
+class EditFile extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      showModal: false
+    }
+  }
+
+  handleEdit = () => {
+    const { activeItems, editFile } = this.props
+    if (activeItems.length === 1 && activeItems[0].type === ITEM_TYPE.folder) {
+      editFolder(activeItems[0].obj.id, this.handleSuccess)
+    }
+    if (activeItems.length === 1 && activeItems[0].type === ITEM_TYPE.file) {
+      editFile(activeItems[0].obj.id, this.handleSuccess)
+    }
+    setActiveItems([])
+  }
+
+  handleSuccess = () => {
+    const id = this.props.currentFolder.id
+    this.props.getFolder(id)
+    this.setState({
+      showModal: false
+    })
+  }
+
+  render () {
+    const { showModal } = this.state
+    const { activeItems } = this.props
+    return (
+      <React.Fragment>
+        <Button
+          icon='edit'
```

Comment:
take this button to bar.js and remove edit-file as there is no use of this file without button